### PR TITLE
Average windspeed triggers park added

### PR DESCRIPTION
**Pull request by mfioren**
The parkAntenna() function gets triggered by the average windspeed rather than the instataneous speed

### CODE REVIEW

**Review by aorlati — APPROVED** [review]

File: Common/Servers/WeatherStation/src/UpdatingThread.cpp [hunk 1]
```diff
@@ -19,11 +25,29 @@ void CUpdatingThread::runLoop()
...
+    if(deq.size() == 10)    
+    	{
+    		wind_avg = deq_sum / deq.size();
+    		ACS_LOG(LM_FULL_INFO,"UpdatingThread::runLoop()",(LM_INFO,"wind avg=%f ", wind_avg));
+    		if(wind_avg > m_threshold) 
+    			{
+       	 		m_weatherstation_p->parkAntenna();
+       	 		ACS_LOG(LM_FULL_INFO,"UpdatingThread::runLoop()",(LM_WARNING,"WINDSPEED_AVG=%f ", wind_avg));
+    			}
+    		deq.erase(deq.begin());
```
Comment (giuseppe-carboni) [inline Common/Servers/WeatherStation/src/UpdatingThread.cpp:42]: Why not use [pop_front](https://en.cppreference.com/w/cpp/container/deque/pop_front)?

File: Common/Servers/WeatherStation/src/UpdatingThread.cpp [hunk 2]
```diff
@@ -19,11 +25,29 @@ void CUpdatingThread::runLoop()
     m_weatherstation_p->updateData();
     Weather::parameters w_pars = m_weatherstation_p->getData();
 
-    if(w_pars.windspeed > m_threshold)
+    deq.push_back(w_pars.windspeed);
+    ACS_LOG(LM_FULL_INFO,"UpdatingThread::runLoop()",(LM_INFO,"Deque size=%zu ", deq.size()));
```
Comment (giuseppe-carboni) [inline Common/Servers/WeatherStation/src/UpdatingThread.cpp:29]: This seems more of a DEBUG level print, not a LM_INFO, as well as...

File: Common/Servers/WeatherStation/src/UpdatingThread.cpp [hunk 3]
```diff
@@ -19,11 +25,29 @@ void CUpdatingThread::runLoop()
...
+    deq_sum  = accumulate(deq.begin(), deq.end(), 0.0);
+    ACS_LOG(LM_FULL_INFO,"UpdatingThread::runLoop()",(LM_INFO,"Deque sum=%f ", deq_sum));
```
Comment (giuseppe-carboni) [inline Common/Servers/WeatherStation/src/UpdatingThread.cpp:31]: ...this...

File: Common/Servers/WeatherStation/src/UpdatingThread.cpp [hunk 4]
```diff
@@ -19,11 +25,29 @@ void CUpdatingThread::runLoop()
...
+    
+    if(deq.size() == 10)    
+    	{
+    		wind_avg = deq_sum / deq.size();
+    		ACS_LOG(LM_FULL_INFO,"UpdatingThread::runLoop()",(LM_INFO,"wind avg=%f ", wind_avg));
```
Comment (giuseppe-carboni) [inline Common/Servers/WeatherStation/src/UpdatingThread.cpp:36]: ...and this

**Review by giuseppe-carboni — CHANGES_REQUESTED** [review]
Some minor tweaks suggested

File: Common/Servers/WeatherStation/src/UpdatingThread.cpp (hunk 1 above)
Comment (mfioren) [reply Common/Servers/WeatherStation/src/UpdatingThread.cpp:42]: Thanks for the advice, I'll try and see if it's faster

File: Common/Servers/WeatherStation/src/UpdatingThread.cpp (hunk 4 above)
Comment (mfioren) [reply Common/Servers/WeatherStation/src/UpdatingThread.cpp:36]: Ok, i'll correct those

**Review by giuseppe-carboni — APPROVED** [review]

**Review by LorenzoMonti — APPROVED** [review]

**Pull request merged**